Multiple variables declaration separated by functions

The following program `sample.go` triggers a panic:

```go
package main

import (
	"fmt"
)

type Foo struct {
	A string
}

var f Foo // <-- the root cause

func Hello() {
	fmt.Println("in")
}

var name = "v1" // <-- the root cause

func main() {
	Hello()
	fmt.Println("Hello", f.A, name)
}
```

Expected result:
```console
$ go run ./sample.go
in
Hello  v1
```

Got:
```console
$ dyngo ./sample.go
in
panic: runtime error: index out of range [recovered]
        panic: runtime error: index out of range

goroutine 1 [running]:
github.com/containous/dyngo/interp.runCfg.func1(0xc0001a85a0)
        /home/ldez/sources/go/src/github.com/containous/dyngo/interp/run.go:97 +0x134
panic(0xbe3840, 0x1644f70)
        /home/ldez/.gvm/gos/go1.11.5/src/runtime/panic.go:513 +0x1b9
github.com/containous/dyngo/interp.callBin.func4(0xc0001a85a0, 0xc0001a8460)
        /home/ldez/sources/go/src/github.com/containous/dyngo/interp/run.go:611 +0x299
github.com/containous/dyngo/interp.runCfg(0xc00020f320, 0xc0001a85a0)
        /home/ldez/sources/go/src/github.com/containous/dyngo/interp/run.go:102 +0x62
github.com/containous/dyngo/interp.(*Interpreter).run(0xc000216070, 0xc00020e360, 0x0)
        /home/ldez/sources/go/src/github.com/containous/dyngo/interp/run.go:84 +0xa6
github.com/containous/dyngo/interp.(*Interpreter).Eval(0xc000216070, 0xc00021a000, 0x106, 0xc9f17f, 0x4, 0xc000216070, 0x0, 0x1157e98)
        /home/ldez/sources/go/src/github.com/containous/dyngo/interp/interp.go:233 +0x48c
main.main()
        /home/ldez/sources/go/src/github.com/containous/dyngo/dyngo.go:50 +0x334
```

--- a/_test/var6.go
+++ b/_test/var6.go
@@ -1,0 +1,26 @@
+package main
+
+import (
+	"fmt"
+)
+
+type Foo struct {
+	A string
+}
+
+var f = Foo{"world"} // <-- the root cause
+
+func Hello() {
+	fmt.Println("in")
+}
+
+var name = "v1" // <-- the root cause
+
+func main() {
+	Hello()
+	fmt.Println("Hello", f.A, name)
+}
+
+// Output:
+// in
+// Hello world v1

--- a/interp/cfg.go
+++ b/interp/cfg.go
@@ -1035,7 +1035,12 @@ func wireChild(n *Node) {
 
 	// Chain sequential operations inside a block (next is right sibling)
 	for i := 1; i < len(n.child); i++ {
-		n.child[i-1].tnext = n.child[i].start
+		switch n.child[i].kind {
+		case FuncDecl:
+			n.child[i-1].tnext = n.child[i]
+		default:
+			n.child[i-1].tnext = n.child[i].start
+		}
 	}
 
 	// Chain subtree next to self

--- a/interp/interp_test.go
+++ b/interp/interp_test.go
@@ -6469,6 +6469,43 @@ func main() {
 	// a: 64 int64
 }
 
+func Example_var6() {
+	src := `
+package main
+
+import (
+	"fmt"
+)
+
+type Foo struct {
+	A string
+}
+
+var f = Foo{"world"} // <-- the root cause
+
+func Hello() {
+	fmt.Println("in")
+}
+
+var name = "v1" // <-- the root cause
+
+func main() {
+	Hello()
+	fmt.Println("Hello", f.A, name)
+}
+`
+	i := interp.New(interp.Opt{Entry: "main"})
+	i.Use(stdlib.Value, stdlib.Type)
+	_, err := i.Eval(src)
+	if err != nil {
+		panic(err)
+	}
+
+	// Output:
+	// in
+	// Hello world v1
+}
+
 func Example_variadic() {
 	src := `
 package main

--- a/interp/run.go
+++ b/interp/run.go
@@ -74,6 +74,7 @@ func (interp *Interpreter) run(n *Node, cf *Frame) {
 	} else {
 		f = &Frame{anc: cf, data: make([]reflect.Value, n.flen)}
 	}
+
 	for i, t := range n.types {
 		// FIXME: nil types are forbidden and should be detected at compile time (CFG)
 		if t != nil && i < len(f.data) && !f.data[i].IsValid() {
@@ -606,7 +607,7 @@ func callBin(n *Node) {
 			}
 			//log.Println(n.index, "callbin", value(f).Type(), in)
 			r := value(f).Call(in)
-			//log.Println(n.index, "callBin, res:", v, fsize, n.findex)
+			//log.Println(n.index, "callBin, res:", r, fsize, n.findex, len(r), len(f.data))
 			for i := 0; i < fsize; i++ {
 				f.data[n.findex+i] = r[i]
 			}

